io/disk/iozone.py

```diff
@@ -0,0 +1,577 @@
+                if match:
+                    # Extract the section name and the worker count
+                    w_count = int(match.group(1))
+                    section = self.__get_section_name(match.group(2))
+
+                    # Output the appropriate keyval pair
+                    key_name = '%s-%d-kids' % (section, w_count)
+                    keylist[key_name] = match.group(3)
+                    continue
+
+                # Check for any other interesting lines
+                if '=' in line:
+                    # Is it something we recognize? First check for parent.
+                    match = parent_regexp.search(line)
+                    if match:
+                        # The section name and the worker count better match
+                        p_count = int(match.group(1))
+                        p_secnt = self.__get_section_name(match.group(2))
+                        if p_secnt != section or p_count != w_count:
+                            continue
+
+                        # Set the base name for the keyval
+                        basekey = 'parent'
+                    else:
+                        # Check for the various 'throughput' values
+                        if line[3:26] == ' throughput per thread ':
+                            basekey = line[0:3]
+                            match_x = kbsec_regexp
+                        else:
+                            # The only other thing we expect is 'Min xfer'
+                            if not line.startswith('Min xfer '):
+                                continue
+                            basekey = 'MinXfer'
+                            match_x = kbval_regexp
+
+                        match = match_x.search(line)
+                        if match:
+                            result = match.group(1)
+                            key_name = "%s-%d-%s" % (section, w_count, basekey)
+                            keylist[key_name] = result
+        output_path = os.path.join(self.outputdir, "perf.json")
+        json.dump(keylist, open(output_path, "w"), indent=1)
+
+    def test(self):
+        '''
+        Test method for performing IOZone test and analysis.
+        '''
+        directory = self.params.get('directory', default=None)
+        args = self.params.get('args', default=None)
+        previous_results = self.params.get('previous_results', default=None)
+
+        if not directory:
+            directory = self.base_dir
+        os.chdir(directory)
+
+        if not args:
+            args = '-a'
+
+        cmd = os.path.join(self.srcdir, 'src', 'current', 'iozone')
+        self.results = process.system_output('%s %s' % (cmd, args))
+        self.auto_mode = ("-a" in args)
+        results_path = os.path.join(self.outputdir,
+                                    'raw_output')
+        analysisdir = os.path.join(self.outputdir,
+                                   'analysis')
+        r_file = open(results_path, 'w')
+        try:
```

Comment:
nitpick: python 2.6 supports `with` statement just for this purpose:
```python
with open(results_path, 'w') as r_file:
    r_file.write(self.results)
```

But this works fine so take it as a suggestion for the future.